### PR TITLE
MINOR: refine DeleteOffsetsConsumerGroupCommandIntegrationTest#produceRecord

### PR DESCRIPTION
**Pull request by FrankYang0529**
Use try-with-resource to create producer and make sure the producer send record without error.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@FrankYang0529 We reuse the TOPIC and GROUP in the for-loop, and that could be a issue if the cleanup of topic/group is not finished before running next loop. Could you please use different topic/group for each loop?

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java [hunk 1]
```diff
@@ -91,65 +91,97 @@ public void testDeleteOffsetsNonExistingGroup() {
...
     @ClusterTest
     public void testDeleteOffsetsOfStableConsumerGroupWithTopicOnly() {
-        for (Map<String, Object> consumerConfig : consumerConfigs) {
-            createTopic(TOPIC);
-            testWithConsumerGroup(TOPIC, -1, 0, Errors.GROUP_SUBSCRIBED_TO_TOPIC, true, consumerConfig);
-            removeTopic(TOPIC);
+        int idx = 0;
+        for (Iterator<Map<String, Object>> it = consumerConfigs.iterator(); it.hasNext(); idx++) {
+            Map<String, Object> consumerConfig = it.next();
+            String topic = TOPIC_PREFIX + idx;
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: Maybe we can use test case + protocol name to be unique name

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java [hunk 2]
```diff
@@ -169,20 +201,22 @@ private static ConsumerGroupCommand.ConsumerGroupService consumerGroupService(St
...
+    private void testWithConsumerGroup(String inputTopicWithData,
+                                       String inputTopicForTest,
+                                       String inputGroup,
                                        int inputPartition,
                                        int expectedPartition,
                                        Errors expectedError,
                                        boolean isStable,
                                        Map<String, Object> consumerConfig) {
-        produceRecord();
+        produceRecord(inputTopicWithData);
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: We can move `produceRecord();` out of `testWithConsumerGroup` to simplify the arguments of `testWithConsumerGroup`

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java (hunk 1 above)
Comment (FrankYang0529) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: Yes, thanks for the suggestion. Updated it.

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java (hunk 2 above)
Comment (FrankYang0529) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteOffsetsConsumerGroupCommandIntegrationTest.java]: I added a function `getValidateRunnable` to separate two topics. Hope it's more clear.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**